### PR TITLE
test: Move common test utilities into `test/utils.js`

### PR DESCRIPTION
**Pull request by ExE-Boss**
Part of: #4688

Extracted from: #4686, see @Elchi3’s comment in https://github.com/mdn/browser-compat-data/pull/4686#issuecomment-524806794 for reasoning.

### CODE REVIEW

File: test/linter/test-browsers.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const browsers = {
  * @param {string[]} displayBrowsers
  * @param {string[]} requiredBrowsers
  * @param {string} category
- * @param {{error:function(...unknown):void}} logger
+ * @param {import('../utils').Logger} logger
```
Comment (Elchi3) [inline test/linter/test-browsers.js]: Is this in preparation to have a logger in utils? I only see a comment about `logger` in utils.js but no implementation. Can you elaborate?

File: test/utils.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+/** Used to check if the process is running in a CI environment. */
+const IS_CI = process.env.CI && String(process.env.CI).toLowerCase() === 'true';
+
+/** Determines if the OS is Windows */
+const IS_WINDOWS = platform() === 'win32';
+
+/**
+ * @param {string} str
+ */
+function escapeInvisibles(str) {
```
Comment (Elchi3) [inline test/utils.js]: It would have been better if a re-implementation of this function would have been done in a separate PR.

File: test/linter/test-browsers.js (hunk 1 above)
Comment (ExE-Boss) [reply test/linter/test-browsers.js]: That’s because the logger is just an object with a defined structure, like an interface in TypeScript.

Comment (Elchi3) [reply test/linter/test-browsers.js]: Ah okay, cool. It looks like it makes sense. Thanks :)
